src/wasm-ast-writer.c

```diff
@@ -335,6 +332,9 @@ static void write_const(Context* ctx, const WasmConst* const_) {
       char buffer[128];
       wasm_write_float_hex(buffer, 128, const_->f32_bits);
       write_puts_space(ctx, buffer);
+      float f32;
```

Comment:
Why not just use a cast, such as the following:
float* f2 = (float*)&const_->f32_bits;

and then writef(..., *f2)

This would avoid the memcpy.